### PR TITLE
Replace become API with (/role/:role_id) or ?role_id= in URLs that use it

### PR DESCRIPTION
**Pull request by Dantemss**
- become API is gone (active roles will be tracked in the url and FE)
- current_role_id in bootstrap data removed (BE doesn't track active roles anymore)

- The following URLs now accept either `/role/:role_id` or `?role_id=`:
  - /api/courses/42/dashboard
  - /api/courses/42/performance(/export(s))
  - /api/courses/42/practice(/worst)
  - /api/courses/42/(teacher_)guide
  - /api/courses/42/notes/8.4 (POST #create ONLY)
  - /api/user/courses/42/student (this is for setting the student_id... you can probably ignore this one, since teacher_students don't have a student_id)

If `role_id` is not provided, whichever of the user's roles was created first in that course is used.

### CODE REVIEW

**Comment by Dantemss** [comment]
Hm I'm not sure... the advantage is that it allows us to tell which routes support the role_id just by looking at `rake routes`, but the disadvantage is that it makes `routes.rb` more complicated than it needs to be. They all have routing tests for both styles of query params though.

File: config/environments/test.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@
   config.action_controller.allow_forgery_protection = false
 
   # Store uploaded files on the local file system in a temporary directory
-  config.active_storage.service = :test
+  # config.active_storage.service = :test
```
[the remark this answers is not in the document]
Comment (Dantemss) [reply config/environments/test.rb]: This was required to disable ActiveStorage. I'd say leave the comment so we know we commented it out next time we upgrade Rails (Rails will probably add this option back in).

File: spec/controllers/api/v1/students_controller_spec.rb [hunk 2]
```diff
@@ -31,72 +33,9 @@
...
-
-  #   context 'caller does not have an authorization token' do
-  #     it 'raises SecurityTransgression' do
-  #       expect {
-  #         api_post :create, nil, params: valid_params, body: valid_body
-  #       }.to raise_error(SecurityTransgression)
-  #     end
-  #   end
-  # end
+  let(:userless_token)    do
```
[the remark this answers is not in the document]
Comment (Dantemss) [reply spec/controllers/api/v1/students_controller_spec.rb]: I think this `ability for a teacher to add a student` is OBE anyway

File: app/subsystems/content/models/page.rb [hunk 3]
```diff
@@ -48,16 +47,18 @@ class Content::Models::Page < IndestructibleRecord
 
   has_many :task_steps, subsystem: :tasks, dependent: :destroy, inverse_of: :page
 
+  has_many :notes, subsystem: :content, dependent: :destroy, inverse_of: :page
+
```
[the remark this answers is not in the document]
Comment (Dantemss) [reply app/subsystems/content/models/page.rb]: We decided to make a future PR and change a bunch of our `dependent: :destroy` to `dependent: :restrict_with_exception`

File: config/environments/test.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (nathanstitt) [reply config/environments/test.rb]: sounds perfect

**Pull request merged**